subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -153,55 +241,38 @@ static int fp_settings_commit(void)
 		return settings_set_err;
 	}
 
-	cur_id = account_key_loaded_ids[0];
-	if ((cur_id != 0) && (account_key_id_to_idx(cur_id) != 0)) {
-		return -EINVAL;
-	}
-
-	if (cur_id == 0) {
-		first_zero_idx = 0;
-	}
-
-	for (size_t i = 1; (i < ACCOUNT_KEY_CNT) &&
-			   (first_zero_idx == -1); i++) {
-		prev_id = cur_id;
-		cur_id = account_key_loaded_ids[i];
-		if (cur_id == 0) {
-			if (prev_id != (i - 1 + ACCOUNT_KEY_MIN_ID)) {
-				return -EINVAL;
-			}
-
+	for (size_t i = 0; i < ACCOUNT_KEY_CNT; i++) {
+		id = account_key_loaded_ids[i];
+		if (id == 0) {
 			first_zero_idx = i;
 			break;
 		}
-		if (cur_id != next_account_key_id(prev_id)) {
-			if (!rollover_check(cur_id, prev_id)) {
-				return -EINVAL;
-			}
 
-			rollover_idx = i;
-			break;
+		if (account_key_id_to_idx(id) != i) {
+			return -EINVAL;
 		}
 	}
 
 	if (first_zero_idx != -1) {
-		if (!zero_id_check(first_zero_idx + 1)) {
-			return -EINVAL;
-		}
+		for (size_t i = 0; i < ACCOUNT_KEY_CNT; i++) {
```

Comment:
Consider splitting this loop into two separate ones with different range (`<0; first_zero_idx)` and `<first_zero_idx; ACCOUNT_KEY_CNT>`). I think it could increase readability.